src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -166,12 +177,10 @@ def test_omp_paralleldo_changes_gen_code():
     assert isinstance(tree.children[0], OMPParallelDoDirective)
     pdir = tree.children[0]
     code = str(psy.gen).lower()
-    assert len(pdir.children) == 4
```

Comment:
I'm sure I should know the answer to this but why does the OMPParallelDoDirective now only have two children where before it had four?

Reply:
The answer may be subtle but it is because before there was some attempt at caching/storing the computed clauses once they were computed (e.g. in the gen_code). I am trying to get rid of these things as much as possible because we don't invalidate the computed values!

So if we modified the tree after a first gen_code and added another private symbol, the next gen_code would not recompute the clauses and before this was generating incorrect code.

For the same reason I am now always recomputing the _get_private_clauses in the lower_to_language_level even if they already exist.

Reply:
Aha. Thanks for explaining.